### DISCUSSION TITLE
Set with Get option?

### DISCUSSION THREAD

**Discussion by tom-lkwd** [redis/rueidis#505] [Q&A]
```
setnxCmd := r.client.B().Set().Key(qualifiedRequestedName).Value(userStr).Nx().ExSeconds(reservationTTLSec).Build()
rx := r.client.Do(ctx, setnxCmd)
```

I add a `.Get()` (as below) and I get an error in my `rueidis.RedisResult` eg `string: "ERR syntax error"` on `Do()` 

```
setnxCmd := r.client.B().Set().Key(qualifiedRequestedName).Value(userStr).Get().Nx().ExSeconds(reservationTTLSec).Build()
rx := r.client.Do(ctx, setnxCmd)
```

I want to `Set if not exists` and if it does exists I want to `Get` the values, is this a bug or am I using this wrong?

**Comment by rueian** [comment]
Hi @tom-lkwd,

The GET option is only available on Redis >= 7.0.

Are you using the latest Redis?
